src/nlsolve/trustRegion.jl

```diff
@@ -97,19 +96,19 @@ function SciMLBase.__solve(prob::NonlinearProblem, alg::SimpleTrustRegion, args.
 
         # Compute the ratio of the actual to predicted reduction.
         @bb Hδ = H × vec(δ)
-        r = (fₖ₊₁ - fₖ) / (dot(δ', g) + dot(δ', Hδ) / T(2))
+        r = (fₖ₊₁ - fₖ) / (dot(δ, g) + dot(δ, Hδ) / T(2))
 
         # Update the trust region radius.
-        if r < η₂
+        if Bool(r ≥ η₂)
```

Comment:
`Bool` is a noop in the general case, but allows me to disambiguate a conditional operation vs a branch in BatchedArrays without messing around with the IR.